### ISSUE TITLE
order of stages on the foyer page

### ISSUE THREAD

**Issue by helenvarley** [upstage-org/upstage#840]
currently the order that stages are displayed on the foyer page is oldest first, newest last. it would be better to have newest first, since this is usually what will be current or upcoming - but even better would be if it's possible to manually control the order. 
could that be part of the "Customise Foyer" options?
copying in @elisadcg & @riadsalameh as it maybe has design implications?

**Comment by riadsalameh** [comment]
Yes it would make more sense to have the newest first. Let me know if there's any changes needed in the design

**Comment by elisadcg** [comment]
I agree newest first :)

**Comment by helenvarley** [comment]
by the way this is connected to #598

**Comment by helenvarley** [comment]
actually it would be good to have the default order newest first, but also to be able to manually order them. it would be logical to have this in the Foyer Customisation setting - a drag & drop list? or numbers that can be changed?

**Comment by vickis** [comment]
Newest for and if possible to override that with a number ranking (for
example you might wish demo stage to be seen first or revisit a work

**Comment by vickis** [comment]
cool - or maybe there could be a pin function on the foyer page ...?

**Comment by helenvarley** [comment]
a manual reordering system would be similar to a pin function but with more control - is this possible?

**Comment by helenvarley** [comment]
we now have the foyer tiles showing with newest first, which is great :slightly_smiling_face: 

is it possible to be able to manually reorder them? i am imagining that this could be done in the "Customise Foyer" interface. could there be thumbnails of the tiles that can be dragged and dropped to a new order and saved?

**Comment by vickis** [comment]
Yes great idea :)

**Comment by tunguyentk** [comment]
Can we make it like this ? 
![image](https://user-images.githubusercontent.com/117065175/217686090-4d8bfbaf-5554-4863-a3b3-e74abbd4122b.png)
But we can sort by what we want by dropmenu (eg newest , oldest , by name (a->z),...) there can also be search by name. The simplest is to place it right above the stages so we can see their changes.

**Comment by vickis** [comment]
Yes sorting by drop down sounds great thank you :)

**Comment by tunguyentk** [comment]
@helenvarley @vickis i made a filter for stage and it work , but im really not good at design , please tell me what i need to change?
![image](https://user-images.githubusercontent.com/117065175/218235121-a7cf371f-bccd-4a8b-9036-7027af3c7396.png)

**Comment by vickis** [comment]
This looks fine to me - probably using capital letters will help with
readability
but @Riad Salameh ***@***.***> or @Elisa Godoy de Castro Guerra
***@***.***> will be better to answer these design questions

great work @tunguyentk :)

**Comment by tunguyentk** [comment]
@vickis I think capitalizing the first letter makes more sense than capitalizing it all.
![image](https://user-images.githubusercontent.com/117065175/218290229-86744ce2-2fbc-4e46-9901-7e2a255e403f.png)
![image](https://user-images.githubusercontent.com/117065175/218290264-bc3de0f0-ec20-4f2f-b11a-d2f7c8c9119a.png)
I have reviewed the other filters of the upstage and capitalizing the first letter will make it more synchronized.
![image](https://user-images.githubusercontent.com/117065175/218290305-81205425-39fd-4711-a074-a0c066c8e804.png)

**Comment by vickis** [comment]
Yes I agree
Oops apologies I just meant the A-Z
Thanks! :)

**Comment by tunguyentk** [comment]
@vickis i deyloyed it. Please check , thanks you so much <3

**Comment by vickis** [comment]
Superb well done! :)

**Comment by helenvarley** [comment]
sorry to chime in late but actually i was not asking for more sorting options, rather to be able to manually re-order the tiles into a custom order.
there will be stages such as the Demo stage which we will want to keep at the top, & sorting alphabetically or chronologically won't achieve that. i can't think of a scenario where we would want to display stages on the foyer alphabetically.
is it possible to have ddrag & drop thumbnails in the Foyer Customisation interface that would allow an Admin to arrange the tiles the way they want?

**Comment by tunguyentk** [comment]
@helenvarley it look really hard for me, It's really hard to do that with the tools currently in use, what I can do now is to add a setting in the settings where you can select a stage from the active stages and always show it at the top

**Comment by tunguyentk** [comment]
How about creating a custom table for priority order? You can sort the order in which stages appear by switching between columns, just like the Player access table.
![image](https://user-images.githubusercontent.com/117065175/218443554-cc42fedb-4669-4418-9a4c-0f7884658ef9.png)

**Comment by vickis** [comment]
Thanks!  :)
That also sounds like a useful solution to stages appearing in the foyer
(keeping the stages in use only and hiding others will help keep the
foyer fresh and in order :)

**Comment by helenvarley** [comment]
yes a table to order the stages would also work. 
however, we don't need to sort which stages appear on the foyer - this is done individually for each stage with the visibility toggle in the stage management interface.
what we need is to have a list of the stages that are set to be visible, and a way to order them.
this should be in the Foyer customisation part of the Admin section.

**Comment by helenvarley** [comment]
hi, i am bumping this up as again it is something that would be a great thing to be able to do. 
i've created a stage for the performances on 8th march, and want it to show at the top of the grid, along with the demo stage, and to have the stages for all four shows and the mobilise one still there on the foyer. at the moment, the new "And Now ... " stage is the last one.

it is definitely a nice-to-have, but if it's not difficult then it would be great to get it done.

**Comment by helenvarley** [comment]
it would be so awesome if we could have this feature in the next release.

i'm imagining it at the end of the Foyer Customisation interface, a row of thumbnails of all the stages that have been set to visible (to appear on the foyer) that can be dragged to reorder. 

is it possible?

**Comment by helenvarley** [comment]
this is not a "must-have" for 2.1.0, but would be great to have if it's not too difficult.

**Comment by helenvarley** [comment]
stages now display in the "most recent activity first" order; so if you want to bring your stage to the top of the list, you just need to have some action on it. closing this issue!

**Comment by helenvarley** [comment]
reopening this issue! as this is not working as expected: i have been active in the demo stage on dev-app2 but it has not come to first place in the foyer tiles.

**Comment by helenvarley** [comment]
this is working now - the stages display on the foyer in order of most recently active.
